### PR TITLE
rgw/tracer: Multipart upload checksum (test attribute draft)

### PR DESCRIPTION
**Pull request by shiyw1209**
rgw/tracer: Multipart upload checksum (test attribute draft)

In this PR I added a test attribute to the source code of raw (multipart upload). At present it's just a test attribute and I have not implement some real function of it yet.
(I mainly changes the source code in `rgw/rgw_tracer.h` and `rgw/rgw_op.cc`.)

It starts on `InitMultipartUpload` Op, and on each `PutObj` Op related to the same multipart upload will be added. So is for the `CompleteMultipartUpload` Op.

I successfully built and run it locally and here is the screenshot.
image
In the tags, a new attribute `upload_checksum` is included.
- It will be initialized to `0` value.

The environment of development:
- Fedora 35 (Linux fedora 5.16.16-200.fc35.x86_64)

## Checklist
- [ ] References tracker ticket
- [ ] Doc update (no ticket needed)

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -6189,6 +6194,11 @@ void RGWInitMultipart::execute(optional_yield y)
     upload_id = upload->get_upload_id();
   }
   s->trace->SetAttribute(tracing::rgw::UPLOAD_ID, upload_id);
+
+  // set an attribute called "upload_id", 
+  // initialize it with value 0 (implement the algo later)
+  s->trace->SetAttribute(tracing::rgw::UPLOAD_CHECKSUM, 0);
+
```
Comment (zenomri) [inline src/rgw/rgw_op.cc:6201]: have you implemented the algo?

Comment (shiyw1209) [reply src/rgw/rgw_op.cc:6201]: No, I haven't. 
I found that there already exists similar attribute like MD5 check in Jaeger.

Comment (zenomri) [reply src/rgw/rgw_op.cc:6201]: ok.
please remove the irrelevant files from the PR. you need to add only `rgw_op.cc` and `rgw_tracer.h`, so the other files that may be generated during the build are not supposed to be part of the PR

Comment (mattbenjamin) [reply src/rgw/rgw_op.cc:6201]: We should really avoid using MD5 for this sort of thing.  The obvious choice would be SHA256 or maybe SHA512 (/2), whichever is faster.  An apparently fast alternative might be Blake2b.

**Comment by shiyw1209** [comment]
Does that mean a Blake2b implementation library needs to be included?

**Pull request closed without merging**